e2e/cypress/integration/enterprise/system_console/system_scheme_permission_spec.js

```diff
@@ -0,0 +1,277 @@
+};
+
+const removePermission = (permissionCheckBoxTestId) => {
+    cy.findByTestId(permissionCheckBoxTestId).then((el) => {
+        if (el.hasClass('checked')) {
+            el.click();
+        }
+    });
+};
+
+// # Checks to see if user recieved a system message warning after using @here
+// # If enabled is true assumes the user has the permission enabled and checks for no system message
+const channelMentionsPermissionCheck = (enabled) => {
+    // # Type @here and post it to the channel
+    cy.get('#post_textbox').clear().type('@here{enter}');
+
+    // # Wait for system message to appear
+    cy.wait(500); // eslint-disable-line cypress/no-unnecessary-waiting
+
+    // # Get last post message text
+    cy.getLastPostId().then((postId) => {
+        if (enabled) {
+            // * Assert that the last message posted is not a system message informing us we are not allowed to use channel mentions
+            cy.get(`#postMessageText_${postId}`).should('not.include.text', 'Channel notifications are disabled');
+        } else {
+            // * Assert that the last message posted is the system message informing us we are not allowed to use channel mentions
+            cy.get(`#postMessageText_${postId}`).should('include.text', 'Channel notifications are disabled');
+        }
+    });
+};
+
+// # Checks to see if the post input is enabled or disalbed and that the API
+// accepts or rejects the create post request.
+const createPostPermissionCheck = (enabled) => {
+    if (enabled) {
+        // # Try post it to the channel
+        cy.get('#post_textbox').should('not.be.disabled');
+        cy.get('#post_textbox').clear().type('test{enter}');
+    } else {
+        // # Ensure the input is disabled
+        cy.get('#post_textbox').should('be.disabled');
+    }
+
+    // # Wait for system message to appear
+    cy.wait(500); // eslint-disable-line cypress/no-unnecessary-waiting
+
+    // # Get last post message text
+    cy.getLastPostId().then((postId) => {
+        if (enabled) {
+            // * Assert that the last message posted is not a system message informing us we are not allowed to use channel mentions
+            cy.get(`#postMessageText_${postId}`).should('include.text', 'test');
+        }
+    });
+};
+
+const resetPermissionsToDefault = () => {
+    // # Login as sysadmin and navigate to system scheme page
+    cy.apiLogin('sysadmin');
+    cy.visit('/admin_console/user_management/permissions/system_scheme');
+
+    // # Click reset to defaults and confirm
+    cy.findByTestId('resetPermissionsToDefault').click();
+    cy.get('#confirmModalButton').click();
+
+    // # Save
+    saveConfig();
+};
+
+describe('System Scheme Channel Mentions Permissions Test', () => {
+    before(() => {
+        // * Check if server has license
+        cy.requireLicense();
+
+        // Reset permissions in system scheme to defaults.
+        resetPermissionsToDefault();
+
+        deleteExistingTeamOverrideSchemes();
+    });
+
```

Comment:
Can you please add an after method to resetPermissionsToDefault to ensure that even if this test fails anywhere in between, the system scheme permissions are reset to defaults and other tests are not affected. 

```
after(() => {

resetPermissionsToDefault();

});